README.md

```diff
@@ -27,7 +27,106 @@
 
 Hazelcast supports EC2 auto-discovery. It is useful when you do not want to provide or you cannot provide the list of possible IP addresses. 
 
-To configure your cluster to use EC2 auto-discovery, follow the below steps:
+There are two possible ways to configure your cluster to use EC2 auto-discovery.
+You can either choose to configure your cluster with `AwsConfig` (or `aws` element in your XML config)
+or you can choose configuring AWS discovery using [Discovery SPI](http://docs.hazelcast.org/docs/latest/manual/html-single/index.html#discovery-spi) implementation.
+Starting with hazelcast-aws-1.2, it's strongly recommended to use Discovery SPI implementation as
+the former one will be deprecated.
+
+###Configuring AWS Discovery using Discovery SPI for Hazelcast Cluster Members
+
+- Add the *hazelcast-aws.jar* dependency to your project. Note that it is also bundled inside *hazelcast-all.jar*. The Hazelcast aws module does not depend on any other third party modules.
+- Disable join over multicast, TCP/IP and aws by setting the `enabled` attribute of the related tags to `false`.
+- Enable Discovery SPI by adding "hazelcast.discovery.enabled" property to your config.
+
+The following is an example declarative configuration.
+
+```xml
+ <hazelcast>
+   ...
+  <properties>
+     <property name="hazelcast.discovery.enabled">true</property>
+  </properties>
+  <network>
+    ...
+    <join>
+        <tcp-ip enabled="false"></tcp-ip>
+        <multicast enabled="false"/>
+        <aws enabled="false" />
+        <discovery-strategies>
+            <!-- class equals to the DiscoveryStrategy not the factory! -->
+            <discovery-strategy enabled="true" class="com.hazelcast.aws.AwsDiscoveryStrategy">
+                <properties>
+                   <property name="access-key">my-access-key</property>
+                   <property name="secret-key">my-secret-key</property>
+                   <property name="iam-role">s3access</property>
+                   <property name="region">us-west-1</property>
+                   <property name="host-header">ec2.amazonaws.com</property>
+                   <property name="security-group-name">hazelcast</property>
+                   <property name="tag-key">aws-test-cluster</property>
+                   <property name="tag-value">cluster1</property>
+                   <property name="hz-port">5701</property>
+                </properties>
+            </discovery-strategy>
+        </discovery-strategies>
+    </join>
+  </network>
+ </hazelcast>
+```
+
+Here are the definitions of properties
+
+* `access-key`, `secret-key`: Access and secret keys of your account on EC2.
+* `iam-role`: If you do not want to use access key and secret key, you can specify `iam-role`. Hazelcast-aws plugin fetches your credentials by using your IAM role. It is optional.
```

Comment:
Is it plugin or module? Again for the sake of consistency, let's choose one and use it for all hazelcast-aws phrases.